__tests__/app.test.js

```diff
@@ -357,5 +353,100 @@ describe('GET /api/users', () => {
+                        author: expect.any(String),
+                        comment_count: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number)
+                    })
+                )
+        })
+            
+       })
+    })
+    test('200: can add a sort_by query and order', () => {
+        return request(app)
+        .get('/api/articles?sort_by=title&order_by=ASC')
+        .expect(200)
+        .then(({body: {articles}})=> {
+            expect(articles).toBeSortedBy('title', {descending: false})
+            articles.forEach((article) => {
+                expect(article).toEqual(
+                    expect.objectContaining({
+                        article_id: expect.any(Number),
+                        title: expect.any(String),
+                        topic: expect.any(String),
+                        author: expect.any(String),
+                        comment_count: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number)
+                    })
+                )
+        })
+            
+       })
+    })
+    test('404: trying to query with a topic that is not yet a topic', () => {
+        return request(app)
+        .get('/api/articles?topic=spheal')
+        .expect(404)
+        .then((response)=> {
+            const msg = response.body.msg;
+            expect(msg).toBe('not found')
+       })
+    })
+    test('400: trying to sort by with a column that is not in articles table should not work', () => {
+        return request(app)
+        .get('/api/articles?sort_by=hitmonlee')
+        .expect(400)
+        .then((response)=> {
+            const msg = response.body.msg;
+            expect(msg).toBe('wrong request')
+       })
+    })
+    test('400: trying to order by with a column that is not desc/asc/DESC/ASC should not work', () => {
+        return request(app)
+        .get('/api/articles?order_by=amaura')
+        .expect(400)
+        .then((response)=> {
+            const msg = response.body.msg;
+            expect(msg).toBe('wrong request')
+       })
+    })
+    test('200: topic does not come up with anything but is a valid topic ', () => {
+        return request(app)
+        .get('/api/articles?topic=paper')
+        .expect(200)
+        .then(({body : {articles}})=> {
+            articles.forEach((article) => {
+                expect(article).toEqual(
+                    expect.objectContaining({
+                        article_id: expect.any(Number),
+                        title: expect.any(String),
+                        topic: expect.any(String),
+                        author: expect.any(String),
+                        comment_count: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number)
+                    })
+                )
+        })
```

Comment:
Good test but I'd argue it doesn't quite just yet tell us that the articles are empty? Maybe add a length check to confirm that there is indeed no articles for this valid topic.